spec/README.md

```diff
@@ -0,0 +1,172 @@
+
+When calculating the rewards, the pool operator takes a cut for each reward that is sent to the pool, where `POOL_FEES_ADDRESS` gets `POOL_FEES_PERCENT`. The rest of the rewards are shared evenly among all eligible validators.
+
+Regarding the pool fees, note that the funds are not sent *per se* to the `POOL_FEES_ADDRESS` but they are added as a leaf in the merkle tree (see merkle tree section). In other words, the owner of the pool can claim the fees as if it were a validator, by providing a valid merkle proof and using said address as sender.
+
+For each reward (see types of rewards) that is sent to the pool on a finalized block, it is distributed as follows:
+* Get the amount of eligible validators (validators that are eligible for rewards) `Active` or `YellowCard` state.
+* The pool takes `POOL_FEES_PERCENT` of that reward, increasing its balance `AccumulatedRewards` by that amount + remainder (if any). Note that all the arithmetic is integer based without decimals, hence the remainder.
+* The reward minus the cut (and the remainder) is shared among all eligible validators. Note that if there is also a reminder, it goes to the `POOL_FEES_ADDRESS`, increasing its `AccumulatedRewards`.
+* Each eligible validator gets its `PendingRewards` increased by that amount.
+
+Note that the pool gets the remainders from two different divisions, but this is done for simplicity and since the calculations are in wei, the value of it is neglectable. Doing this makes the oracle fair with all validators, since each one of them gets the exact same amount of rewards. So in practice, `POOL_FEES_ADDRESS` just gets `POOL_FEES_PERCENT`.
+
+TODO: Draw diagram with flow.
+TODO: Include subscriptions and unsubscriptions
+
+## Merkle trees and proofs
+
+Since storing all rewards calculations on-chain would be almost impossible and very expensive, merkle trees are used to summarize the state of all validators tracked by the oracle in a given value called **merkle root*. All the computation of the rewards is done off-chain by the oracle, and on every `CHECKPOINT_SIZE_SLOTS` all rewards all calculated and summarized in a new merkle root that is stored on-chain in Ethereum.
+
+Each leaf of the tree contains two values, the deposit address and the accumulated balance. Note that in order to be more gas efficient and allow to claim of multiple validators in just one transactions, all validators belonging to the same deposit address are aggregated. Meaning that if deposit address `da1` deposited validators `v1`, `v2` and `v3` and all of them are subscribed to the pool with respective accumulated balances of `a1`, `a2` and `a3`, then this is represented as a single leaf in the tree where `da1` and `v1+v2+v3`.
```

Comment:
The entire readme needs to be updated for the deposit address --> withdrawal address change.